src/transformers/pipelines/zero_shot_object_detection.py

```diff
@@ -0,0 +1,134 @@
+                - An image loaded in PIL directly
+
+            candidate_labels (`List[str]`):
+                The candidate labels for this image
+
+            hypothesis_template (`str`, *optional*, defaults to `"This is a photo of {}"`):
+                The sentence used in cunjunction with *candidate_labels* for assigning labels to predicted bounding
+                boxes by replacing the placeholder with the candidate_labels. Then likelihood is estimated by using
+                logits
+
+        Return:
+            A list of dictionaries containing result, one dictionary per proposed label. The dictionaries contain the
+            following keys:
+
+            - **label** (`str`) -- The label identified by the model. It is one of the suggested `candidate_label`.
+            - **score** (`float`) -- The score attributed by the model for that label (between 0 and 1).
+            - **box** (`List[Dict[str,int]]`) -- The bounding box of detected object in image's original size.
+        """
+        return super().__call__(images, **kwargs)
+
+    def _sanitize_parameters(self, **kwargs):
+        preprocess_params = {}
+        if "candidate_labels" in kwargs:
+            preprocess_params["candidate_labels"] = kwargs["candidate_labels"]
+        if "hypothesis_template" in kwargs:
+            preprocess_params["hypothesis_template"] = kwargs["hypothesis_template"]
+
+        postprocess_params = {}
+        if "threshold" in kwargs:
+            postprocess_params["threshold"] = kwargs["threshold"]
+        return preprocess_params, {}, postprocess_params
+
+    def preprocess(self, image, candidate_labels=[None], hypothesis_template="This is a photo of {}."):
+        image = load_image(image)
+        target_size = torch.IntTensor([[image.height, image.width]])
+        images = self.feature_extractor(images=[image], return_tensors=self.framework)
+        inputs = self.tokenizer(
+            [hypothesis_template.format(label) for label in candidate_labels],
+            return_tensors=self.framework,
+            padding=True,
+        )
+        inputs["pixel_values"] = images.pixel_values
+        inputs["target_size"] = target_size
+        return {"candidate_labels": candidate_labels, "target_size": target_size, **inputs}
+
+    def _forward(self, model_inputs):
+        target_size = model_inputs.pop("target_size")
+        candidate_labels = model_inputs.pop("candidate_labels")
+        outputs = self.model(**model_inputs)
+
+        model_outputs = outputs.__class__(
+            {"target_size": target_size, "candidate_labels": candidate_labels, **outputs}
+        )
+        return model_outputs
+
+    def postprocess(self, model_outputs, threshold=0.1):
+        text = model_outputs["candidate_labels"]
+
+        results = self.feature_extractor.post_process(outputs=model_outputs, target_sizes=model_outputs["target_size"])
+        keep = results[0]["scores"] >= threshold
```

Comment:
can we avoid the hardcoded `[0]` and use `more_itertools.one` instead? More explicit